base/system/winlogon/sas.c

```diff
@@ -1344,38 +1497,142 @@ UnregisterHotKeys(
     return TRUE;
 }
 
-BOOL
-WINAPI
-HandleMessageBeep(UINT uType)
+BOOL WINAPI HandleMessageBeep(PWLSESSION Session, UINT uType)
 {
-    LPWSTR EventName;
-
-    switch(uType)
-    {
-    case 0xFFFFFFFF:
-        EventName = NULL;
-        break;
-    case MB_OK:
-        EventName = L"SystemDefault";
-        break;
-    case MB_ICONASTERISK:
-        EventName = L"SystemAsterisk";
-        break;
-    case MB_ICONEXCLAMATION:
-        EventName = L"SystemExclamation";
-        break;
-    case MB_ICONHAND:
-        EventName = L"SystemHand";
-        break;
-    case MB_ICONQUESTION:
-        EventName = L"SystemQuestion";
-        break;
-    default:
-        WARN("Unhandled type %d\n", uType);
-        EventName = L"SystemDefault";
-    }
-
-    return PlaySoundRoutine(EventName, FALSE, SND_ALIAS | SND_NOWAIT | SND_NOSTOP | SND_ASYNC);
```

Comment:
What's the rationale behind why it's not possible to directly use the PlaySoundRoutine as is?

Reply:
PlaySound function with SND_ASYNC flag calls CreateThread function that doesn't work with user impersonation.